Composer/packages/lib/shared/src/lgUtils/models/LgTemplateRef.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+import parseLgTemplateRef from '../parsers/parseLgTemplateRef';
+
+import { LgTemplateName, LgTemplateRefString, LgText } from './stringTypes';
+
+export default class LgTemplateRef {
+  name: LgTemplateName;
+
+  parameters: string[] | undefined;
```

Comment:
is it OK to just typed as string[] without `undefined`?

Reply:
because `[template()]` is valid, and in the following update `@{template()}` will require a `()`

Reply:
parameters = `undefined` will produce `greeting` while prarameters = [] will produce `greeting()`. If we don't want the parameters be undefined, defence it in constructor?

Reply:
yes, i see what you are doing in toString(), my point was greating is not valid, and greating() is always valid, so we don't need cases like 'greating', because we call this "lgTemplateRef".

Reply:
What about those templates created by Composer? like `bfdactivity-815412`, should it be `[bfdactivity-815412]` or `[bfdactivity-815412()]` in .dialog file? @boydc2014

Reply:
my understand is it should be consistent with lg grammar which means we need to update all sample bots

Reply:
So i want to make this change: 1. let composer generate strings like `[bfdxxx-xxx()]` 2. but don't update sample bots since they still work, let's update them in the PR which updates lg version (as a breaking change)

Reply:
done